src/components/Modal/CategoryListModal/CategoryList.tsx

```diff
@@ -1,23 +1,34 @@
 import Button from '@components/Button';
 import styled, { css } from 'styled-components';
 import { Category } from 'types/category';
+import { usePostProductModalStore } from '../PostProductModal/usePostProductModalStore';
 
 type Props = {
   categories: Category[];
-  onClick: (category: Category) => void;
 };
 
-export const CategoryList: React.FC<Props> = ({ categories, onClick }) => {
+export const CategoryList: React.FC<Props> = ({ categories }) => {
+  const setSelectCategory = usePostProductModalStore(
+    ({ setSelectCategory }) => setSelectCategory,
+  );
+  const closeCategoryListModal = usePostProductModalStore(
+    ({ closeCategoryListModal }) => closeCategoryListModal,
+  );
```

Comment:
[의견]
슬슬 하나의 훅에서 꺼내 쓸 때 훅 실행을 여러번 해야하나? 하는 생각이 들 것 같아요. 아래에서는 훅실행을 4번도 더 한 파일이 있는 것 같더라구요.
https://github.com/pmndrs/zustand/discussions/1449
해당 디스커션을 참고해보시는 것은 어떨까요?

Reply:
아앗... 같은 store를 여러번 호출하지 않아도 되는 방법이 있었군요! 최근 zustand에 useShallow라는 훅이 추가되어서 확인해보니 잘 동작합니다. 아래처럼 개선해볼 수 있을 것 같아요. 👍👍

```suggestion
const setSelectCategory = usePostProductModalStore(
    useShallow(({ setSelectCategory, closeCategoryListModal }) => ({
      setSelectCategory,
      closeCategoryListModal,
    })),
  );
```